x-pack/plugins/security/server/audit/audit_service.ts

```diff
@@ -101,6 +104,20 @@ export class AuditService {
       )
     );
 
+    // Record feature usage at a regular interval if enabled and license allows
+    if (config.enabled && config.appender) {
+      license.features$.subscribe((features) => {
+        clearInterval(this.usageIntervalId!);
+        if (features.allowAuditLogging) {
+          recordAuditLoggingUsage();
+          this.usageIntervalId = setInterval(recordAuditLoggingUsage, RECORD_USAGE_INTERVAL);
+          if (this.usageIntervalId.unref) {
```

Comment:
If I understand correctly, we only need this guard because the jest tests are running witin the `jsdom` [environment](https://jestjs.io/docs/en/configuration#testenvironment-string) (as defined in https://github.com/elastic/kibana/blob/b593781009c47b4f5cf8d5010d571d55f6b2dcbc/packages/kbn-test/jest.config.js).

@tylersmalley have there been other folks asking for different jest test environments? I could see us supplying a different environment depending on where the test is located: `public` and `client` could use the existing `jsdom`-based environment, while tests within the `server` directory could use the `node` environment. Happy to open an issue to discuss further. It's certainly not a blocker for this PR.

Reply:
No asks thus far - but you can create a different `jest.config.js` file for `server` and `public`. We would need to update the [projects match](https://github.com/elastic/kibana/blob/master/x-pack/jest.config.js#L10).

Reply:
Thanks for the pointer!

Reply:
No problem, let me know if you would like help with it.

Reply:
You can switch node environment using comment at the top of your test file:

```
/**
 * @jest-environment node
 */
```

Unfortunately that throws an errors in our case since some of the polyfills imported in our jest config rely on window object, which is why I added the guard in the code instead, as it seemed a bit of a can of worms. Would be great to be able to use node environment in jest for server/ tests though to get them closer to production env.